src/Tag/Tag.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { Clickable, CompositeItem, CompositeStateReturn } from "reakit";
+
+import theme from "../theme";
+import { ocx } from "../utils";
+import { CrossIcon } from "../icons";
+import { forwardRefWithAs, PropsWithAs } from "../utils/types";
+
+export const TagsContext = React.createContext<CompositeStateReturn | null>(
+  null,
+);
+export const useTagsContext = () => {
+  return React.useContext(TagsContext);
+};
+
+export type TagProps = {
+  /**
+   * id for the tag
+   */
+  id?: string;
+  /**
+   * Is tag closable?
+   */
+  closable?: boolean;
+  /**
+   * Callback to fire when close icon is clicked?
+   */
+  onClose?: (id?: string) => void;
+  /**
+   * How large should the button be?
+   */
+  size?: "xs" | "sm" | "md" | "lg";
+  /**
+   * If added, tag will show prefix the content before the tag's label
+   */
+  prefix?: React.ReactElement;
+  /**
+   * If added, tag will show suffix after the tag's label.
+   * NOTE: This will only show if closable is set to `true`
+   */
+  suffix?: React.ReactElement;
+};
+
+function TagComponent(
+  props: PropsWithAs<TagProps, "span">,
+  ref: React.Ref<HTMLSpanElement>,
+) {
+  const {
+    id,
+    size = "md",
+    prefix,
+    suffix = <CrossIcon />,
+    className,
+    closable,
+    onClose,
+    children,
+    ...rest
+  } = props;
+  const tagStyles = ocx(theme.tag.base, theme.tag.size[size], className);
+
+  // TODO: Clean this up
+  if (!closable && suffix.type.displayName !== (CrossIcon as any).displayName) {
+    console.warn(
+      "Tag: `suffix` will not be visible because `closable` is set to false, please set `closable` to true",
+    );
+  }
+
+  const TagWithPrefixSuffix = () => (
+    <>
+      {prefix && <span className={theme.tag.prefix}>{prefix}</span>}
+      {children}
+      {closable && suffix && (
```

Comment:
If not `closable` suffix, can just render in a span would be enough !

Reply:
If closable is false it would not render anything.

Reply:
What if someone what some other icon or avatar in suffix but doesn't want to be a closable.

Reply:
I think we can keep it as it is, I don't see any use of other kind of icons in suffic for tag.

Reply:
> What if someone what some other icon or avatar in suffix but doesn't want to be a closable.


This component is not for that use case then, they can use the badge for that since badge does not handle any Action button logics. 

And ideally prefix will have the Avatar icons and postfix will have the action button.
see https://github.com/timelessco/renderlesskit-react-tailwind/pull/12#discussion_r544844260